server/src/main/java/io/druid/segment/loading/OmniSegmentLoader.java

```diff
@@ -52,12 +54,14 @@
   public OmniSegmentLoader(
```

Comment:
Currently the OmniSegmentLoader handles a lot of functionality around cache management, so things like:

1. Manage choosing index cache locations
1. Manage cleaning up index cache locations
1. Manage download coordination of caches (downloadMarker)
1. Primary implementation of `io.druid.segment.loading.SegmentLoader`

So effectively, the OmniSegmentLoader becomes the "SegmentLocalCacheManager"